subprojects/core/build.gradle.kts

```diff
@@ -314,7 +314,11 @@ tasks.test {
 }
 
 tasks.compileTestGroovy {
-    groovyOptions.fork("memoryInitialSize" to "128M", "memoryMaximumSize" to "1G")
+    groovyOptions.isFork = true
+    groovyOptions.forkOptions.run {
```

Comment:
This seems like a poor workaround. Can we create an issue to introduce `forkOptions(Action)` and similar, perhaps for this same release?

Reply:
If we wanted to add an action-accepting method like that I would probably include it in this PR. 

However, traditionally action-accepting methods called X usually _only_ apply an action to the nested property called _X_. I think it would be unexpected, although convenient, for this to also set `fork = true` on the parent options

Reply:
If it's called `fork` instead I don't see an issue with that.

Reply:
Spoke with sterling and we decided not to do this

Reply:
Lets add an action-accepting method for forkOptions that does not enable forking. 
Lets make it non-incubating

Also for debugOptions